pages/docs/reference/whatsnew1430.md

```diff
@@ -0,0 +1,473 @@
+Kotlin is introducing experimental record class support.
+
+You can use record classes that are declared in Java just like classes with properties in Kotlin. No additional steps are required.
+
+Starting with 1.4.30, you can declare the record class in Kotlin using the `@JvmRecord` annotation for a [data class](data-classes.html):
+
+<div class="sample" markdown="1" mode="kotlin" theme="idea" data-highlight-only>
+
+```kotlin
+@JvmRecord
+data class User(val name: String, val age: Int)
+```
+</div>
+
+To try the preview version of JVM records, add the compiler options `-Xjvm-enable-preview` and `-language-version 1.5`.
+
+We’re continuing to work on JVM records support and we’d be very grateful if you would share your feedback with us using
+this [YouTrack ticket](https://youtrack.jetbrains.com/issue/KT-42430).
+
+Learn more about implementation, restrictions, and the syntax in [KEEP](https://github.com/Kotlin/KEEP/blob/master/proposals/jvm-records.md).
+
+### Sealed interfaces
+
+> Sealed interfaces are [Experimental](evolution/components-stability.html). They may be dropped or changed at any time.
+> Opt-in is required (see the details below), and you should use them only for evaluation purposes. We would appreciate your feedback on them in [YouTrack](https://youtrack.jetbrains.com/issue/KT-42433).
+{:.note}
+
+In Kotlin 1.4.30, we’re shipping the prototype of _sealed interfaces_. They complement sealed classes and make it possible
+to build more flexible restricted class hierarchies.
+
+They can serve as “internal” interfaces that cannot be implemented outside the same module. You can rely on that fact,
+for example, to write exhaustive `when` expressions.
+
+<div class="sample" markdown="1" mode="kotlin" theme="idea" data-highlight-only>
+
+```kotlin
+sealed interface Polygon
+
+class Rectangle(): Polygon
+class Triangle(): Polygon
+
+// when() is exhaustive: no other Polygon implementations can appear
+// after the module is compiled
+fun draw(polygon: Polygon) = when (polygon) {
+    is Rectangle -> // ...
+    is Triangle -> // ...
+}
+```
+</div>
+
+Another use-case: with sealed interfaces, you can inherit a class from two or more sealed superclasses.
+
+<div class="sample" markdown="1" mode="kotlin" theme="idea" data-highlight-only>
+
+```kotlin
+sealed interface Fillable {
+   fun fill()
+}
+sealed interface Polygon {
+   val vertices: List<Point>
+}
+
+class Rectangle(override val vertices: List<Point>): Fillable, Polygon {
+   override fun fill() { /*...*/ }
+}
+```
+</div>
+
+To try the preview version of sealed interfaces, add the compiler option `-language-version 1.5`. Once you switch to this
+version, you’ll be able to use the `sealed` modifier on interfaces. We’d be very grateful if you would share your feedback
+with us using this [YouTrack ticket](https://youtrack.jetbrains.com/issue/KT-42433).
+
+[Learn more about sealed interfaces](sealed-classes.html).
```

Comment:
Don't you want to add a link to a specific section about sealed interfaces?

Reply:
Yeah, sure. I just couldn't do it before the doc update is merged. And the same for the link in the next section.